src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -390,8 +400,15 @@ private static void cleanUp() {
     }
     Logger.LOG_CRITICAL(CONNECTOR_FRIENDLY_NAME + " has finished running.");
 
-    // Disable automatic application restart
-    SCAppManagement.disableAppAutoRestart();
+    // Restart connector if requested, otherwise disable automatic application restart
+    if (restartAppAfterShutdown) {
+      // Exit with non-zero exit code otherwise app auto restart doesnt work
```

Comment:
Missing an apostrophe in doesn't 
ah... I didn't realize that the auto-restart requires a non zero return value.  That makes sense.

Reply:
The documentation says it works with normal and non-normal exit codes, but during testing, I could only make it work with non-normal exit codes.